File: openmeter/server/router/info.go
package router

import "net/http"

// List supported currencies
// (GET /api/v1/info/currencies)
func (a *Router) ListCurrencies(w http.ResponseWriter, r *http.Request) {
	a.infoHandler.ListCurrencies().ServeHTTP(w, r)
}

// Get progress
// (GET /api/v1/info/progress/{progress_id})
func (a *Router) GetProgress(w http.ResponseWriter, r *http.Request, progressID string) {
	a.progressHandler.GetProgress().With(progressID).ServeHTTP(w, r)
}

File: api/client/javascript/src/client/info.ts
import type { Client } from 'openapi-fetch'
import type { RequestOptions } from './common.js'
import type { operations, paths } from './schemas.js'
import { transformResponse } from './utils.js'

/**
 * Info utilities for OpenMeter
 */
export class Info {
  constructor(private client: Client<paths, `${string}/${string}`>) {}

  /**
   * List supported currencies
   * @description List all supported currencies.
   * @param options - The request options
   * @returns The supported currencies
   */
  public async listCurrencies(options?: RequestOptions) {
    const resp = await this.client.GET('/api/v1/info/currencies', {
      ...options,
    })

    return transformResponse(resp)
  }

  /**
   * Get progress
   * @param id - The ID of the progress to get
   * @param options - The request options
   * @returns The progress
   */
  public async getProgress(
    id: operations['getProgress']['parameters']['path']['id'],
    options?: RequestOptions,
  ) {
    const resp = await this.client.GET('/api/v1/info/progress/{id}', {
      params: {
        path: { id },
      },
      ...options,
    })

    return transformResponse(resp)
  }
}
